### DISCUSSION TITLE
컨벤션 규칙

### DISCUSSION THREAD

**Discussion by realswxx** [Tudy-Project/TUDY#4] [공지]
### 1. 이슈 등록 컨벤션
>제목 : [Feat]~~~
내용 : - [ ] 체크리스트 

### 2. 브랜치 네임 컨벤션
>feat/~~

### 3. 커밋 컨벤션
>head : [Feat] 간략 요약 - #123 
body :  설명 

>추가 내용 :
--- Head ---
[Type] Message - #이슈번호
Title은 최대 50글자 정도로만 입력
Title 끝에 마침표 금지, 무엇을 했는지 명확하게 작성
ex) [Feat] message - #123

>--- Body ---
>- 상세 설명

>--- Type ---
Feat    : 새로운 기능 추가
Fix     : 버그 수정
Style   : 코드 의미에 영향을 주지 않는 변경사항 (세미콜론, 줄 간격 등등)
Design  : Storyboard, UI 수정
Refactor: 코드 리팩토링
Test    : 모든 테스트 작업
Docs    : 문서 추가 및 수정
Chore   : 빌드 부분, 패키지 매니저, 세팅 관련 파일 추가 및 수정
Rename  : 파일 네이밍 변경
Remove  : 파일 제거
!HOTFIX  : 빠른 버그수정 필요(해당 내용이 오류일 경우 앱 크래시 증상 등등)

### 3-1. PR 및 코드 리뷰 관련 커밋 컨벤션
>1. 리뷰사항 반영 시
[Chore] 리뷰사항 반영
>2. 컨플릭트 잡을 때
[Chore] Conflict fixed

### 4. Example
>git commit
.gitmessage.txt
git commit -m “ HEAD “

>head
[Feat] 로그인 구현하기 - #2

>body
-파이어베이스 연결 
-카카오톡 연결

### 5. 깃허브 컨벤션
>- 매일 저녁 8시까지 Dev에다가 PR
 -- (8시까지 PR각자 올리고, 팀원 코드 본 후, 8시반부터 코드리뷰 같이 진행) // 올릴게 없는 경우 올린 사람만 진행
>- 매주 목요일 Main에 Dev merge(1주일에 한번 , 한 분이)

### 6. 스냅킷 컨벤션
> 1.top 2.bottom 3.leading 4.trailing
5.width 6.height 7.centerX 8.centerY

### 7. VC 컨벤션 
> 1. VC 한 파일로 해결 (ViewController+UI.swift 만들 필요 X)
> 2. VC class에는 properties / life cycle
> 3. 그 파일 밑에 extension으로 UI/delegate 등 다 빼기
> 4. layout잡는 함수는 configureUI()
> 5. delegate / datasource도 밑에 extension 하나 더 만들어서 작성
> 6. //MARK: -  <- properties / life cycle / extension으로 적어주면 됨

**Comment by change-challenge** [comment]
VC컨벤션 추가했습니다!
